drivers/I2C.h

```diff
@@ -117,8 +117,8 @@ class I2C {
      *  @param repeated Repeated start, true - do not send stop at end
      *
      *  @returns
-     *      0 or non-zero - written number of bytes,
-     *      negative - I2C_ERROR_XXX status
+     *       0 on success (ack),
+     *   non-0 on failure (nack)
```

Comment:
Well, there a two different methods/functions: one is **method** `I2C::write(int, const char*, int, bool)` declared in `drivers/I2C.h`, the other one is **function** `i2c_write(i2c_t*, int, const char, int, int)' declared in `hal/i2c_api.h`.
Method `I2C::write(int, const char*, int, bool)` uses function `i2c_write(i2c_t*, int, const char, int, int)' to implement its functionality (see [here](https://github.com/ARMmbed/mbed-os/blob/master/drivers/I2C.cpp#L60-L70)). In particular note that the return value of method `I2C::write(int, const char*, int, bool)` is computed in this [line](https://github.com/ARMmbed/mbed-os/blob/master/drivers/I2C.cpp#L69) ...